src/feature_annotation/score_genes_cell_cycle_scanpy/config.vsh.yaml

```diff
@@ -0,0 +1,81 @@
+functionality:
+  name: score_genes_cell_cycle_scanpy
+  namespace: feature_annotation
+  description: |
+    Calculates the score associated to S phase and G2M phase and annotates the cell cycle phase for each cell, as implemented by scanpy. 
+    The score is the average expression of a set of genes subtracted with the average expression of a reference set of genes.
+  argument_groups:
+
+    - name: Inputs
+      arguments:
+        - name: --input
+          type: file
+          description: Input h5mu file
+          required: true
+          example: input_file.h5mu
+        - name: --s_genes
+          type: file
+          description: |
+            Path to a .txt file containing the gene list of s phase genes to be scored. 
+            The gene list file should be formatted as a single column with gene symbols.
+          required: true
+          example: gene_list.txt
+        - name: --g2m_genes
+          type: file
+          description: |
+            Path to a .txt file containing the gene list of g2m phase genes to be scored. 
+            The gene list file should be formatted as a single column with gene symbols.
+          required: true
+          example: gene_list.txt
+        - name: --gene_pool
+          type: file
+          description: |
+            File with genes for sampling the reference set. Default is all genes. 
+            The gene pool file should be formatted as a single column with gene symbols.
+          required: false
+          example: gene_pool.txt
+
+    - name: Outputs
+      arguments:
+        - name: --output
+          type: file
+          direction: output
+          description: |
+            Output h5mu file
+          required: true
+          example: output_file.h5mu
+
+    - name: Arguments
+      arguments:
+        - name: --n_bins
+          type: integer
+          default: 25
+          required: false
+          description: |
+            Number of expression level bins for sampling.
+        - name: --random_state
+          type: integer
+          description: |
+            The random seed for sampling.
+          default: 0
+          required: false
+        - name: --use_raw
```

Comment:
In OP, we chose to handle layers a bit different to what scanpy does.  Scanpy will by default use the `.X` layer and overwrite it. If `--use_raw` is specified, it uses `.raw`. In openpipelines we try to do two things: (1) in workflows we do not overwrite layers when they do not hold the same data (2) A user should be able to select from more input layers than `.X` or `.raw`. This means that in worfklows .X is the non-transformed data (not log-normalized) and we add layers `.layers['log_normalized']` for example. This eliminates the use of the `.raw` attribute (because we store it at .X), and we add an `input_layer` argument to the components. TLDR: could you replace `--use_raw` with an `--input_layer` argument (if not specified, use .X , and if specified use `.layers[par['input_layer']`)?

Reply:
As a consequense of this, you might need to create a 'dummy' AnnData file to pass to scanpy, with a copy of the input layer specified by `--input_layer` located at `.X`